internal/pkg/watch/docker_logs.go

```diff
@@ -145,6 +143,11 @@ func (w *DockerLogWatch) StartUnsafe() {
 	)
 
 	newEventStream := func() bool {
+		// Before initializing a log stream, confirm that we need a log watcher
+		if !global.BlockchainNode.LogWatchEnabled() {
```

Comment:
If node hasn't been discovered yet, how will the log watch be enabled later when discovery happens?

Reply:
I'm not sure if I fully understand the question.

Inside the [LogWatchEnabled()](https://github.com/Metrika-Inc/agent/pull/95/files#diff-da8a9dabe9d8327316ce10f36b13721c77948b102cba4674c2a590c8d10de5a5R504) function there's the logic to be looking at the log stream if the node is undiscovered or consensus. Do you think that's not enough to cover the case?

Reply:
Right, this is going to work for consensus nodes.

Now that got me thinking how will the log watch stop once nodeRole is discovered for a non-consensus node. 🤔